client/modules/IDE/components/NewFolderModal.jsx

```diff
@@ -32,7 +32,17 @@ NewFolderModal.propTypes = {
   closeModal: PropTypes.func.isRequired
 };
 
+function validate(formProps) {
+  const errors = {};
+
+  if (!formProps.name || !formProps.name.replace(/\s/g, '').length) {
```

Comment:
rather than using a RegEx here, I would use [String.prototype.trim()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim).